### ISSUE TITLE
Support for FreeBSD

### ISSUE THREAD

**Issue by ymitsos** [laurent22/joplin#867]
Hi,

FreeBSD is not natively supported in the console version of joplin. Seems that a patch of minor significance and complexity in ./ReactNativeClient/lib/shim.js makes it work. Was wondering if a such PR would make sense and is acceptable.

**Comment by foxmask** [comment]
It's related to Electron [here is a request](https://github.com/electron/electron/issues/3797)

**Comment by ymitsos** [comment]
again, I am not claiming expertise, however this very dummy patch works for me. 
```
diff .joplin-bin/lib/node_modules/joplin/lib/shim.js ~/repos/joplin/ReactNativeClient/lib/shim.js
14d13
<         console.info(process && process.platform)
18,22d16
< shim.isFreeBSD = () => {
<         console.info(process && process.platform)
<       return process && process.platform === 'freebsd';
< }
< 
35,36c29
<       if (shim.isLinux()) return 'linux';
<       if (shim.isFreeBSD()) return 'freebsd';
---
```
I am closing the issue.

**Comment by laurent22** [comment]
@ymitsos, you've added a new function "isFreeBSD", but where would it be used? Just trying to understand so that your patch can be integrated.

**Comment by ymitsos** [comment]
@laurent22 in the ~/repos/joplin/ReactNativeClient/lib/shim.js as per the above patch

**Comment by laurent22** [comment]
Thanks, I see why you made this change now. I'm re-opening the issue as a reminder for myself to add the patch.

**Comment by ymitsos** [comment]
I can send you a PR if you prefer so. Thank you anyway.

**Comment by laurent22** [comment]
Yes if you could send a PR that would be great too.

**Comment by ymitsos** [comment]
addressed in https://github.com/laurent22/joplin/pull/896

**Comment by laurent22** [comment]
Fixed in #896
